src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+        except Exception:
+            raise ValueError("Required child MirrorDescanner not provided")
+        self._MirrorDescanner = MirrorDescanner("MirrorDescanner", role=None, parent=self)
+        self.children.value.add(self._MirrorDescanner)
+
+        try:
+            kwargs = children["EBeamScanner"]
+        except Exception:
+            raise ValueError("Required child EBeamScanner not provided")
+        self._EBeamScanner = EBeamScanner("EBeamScanner", role=None, parent=self)
+        self.children.value.add(self._EBeamScanner)
+
+        try:
+            kwargs = children["MPPC"]
+        except Exception:
+            raise ValueError("Required child MPPC not provided")
+        self._MPPC = MPPC("MPPC", None, self)
+        self.children.value.add(self._MPPC)
+
+    @classmethod
+    def simpleASMCall(cls, url, method, expected_status, data=None, raw_response=False, timeout=600):
+        """
+
+        :param url: url of the command, server part is defined in global variable URL
+        :param method: getting or posting via global variables _METHOD_GET/_METHOD_POST
+        :param expected_status: expected feedback of server for a positive call
+        :param data: data added to the call
+        :param raw_response: specified the format of the structure returned
+        :param timeout: [s] if within this period no bytes are received an timeout exception is raised
+        :return: status_code(posting), content dictionary(getting), or entire response (raw_response=True)
+        """
+        logging.getLogger().setLevel(logging.INFO)
+        logging.info("Executing: %s" % url)
+        if method == _METHOD_GET:
+            resp = requests.get(url, json=data, timeout=timeout)
+        elif method == _METHOD_POST:
+            resp = requests.post(url, json=data, timeout=timeout)
+
+        if resp.status_code != expected_status:
+            raise AsmApiException(url, resp, expected_status)
+
+        logging.info("Call to %s went fine, no problems occured\n" % url)
+
+        if raw_response:
+            return resp
+        elif method == _METHOD_POST:
+            return resp.status_code
+        elif method == _METHOD_GET:
+            return json.loads(resp.content)
+
+
+class EBeamScanner(model.Emitter):
+
+    def __init__(self, name, role, parent, **kwargs):
+        super(EBeamScanner, self).__init__(name, role, **kwargs)
+        self.parent = parent
+
+        # self._metadata[model.MD_PIXEL_SIZE] =
+
+        clockFrequency = self.parent.simpleASMCall(self.parent.server_URL + "/scan/clock_frequency", _METHOD_GET, 200)[
+            'frequency']
+        self.clockFrequency = model.IntVA(clockFrequency, unit='Hz')
+        self.clockPeriod = model.FloatVA(1 / self.clockFrequency.value, unit='s')
+        self._shape = model.TupleVA((6400, 6400), unit='px,px')
+        self.resolution = model.ResolutionVA((6400, 6400), ((10, 10), (7200, 7200)), setter=self._setResolution)
```

Comment:
Actually it is defined by the API, the max_resolution = 900(x_cell_size, max) * mmpc._shape.
The minimum of x_cell_size = 10 so that minmimum is defined not that nice now, ideally that is dependend on the mppc parameter. But that is defined later...


x_eff_cell_size = self.parent._eBeam_scanner.resolution.value[0] / self._shape.value[0]

Reply:
by deriving the maximum effective cell size defined in the API I came to this number, comment is added.

Reply:
The example in the API is 900. A maximum was not defined.
        x_cell_size:
          description: "Length of the scan line (x direction) in a single cell image in pixels, including overscan pixels."
          type: integer
          format: int64
          minimum: 10
          example: 900

Reply:
How about a max of 1000 * 8? I think that we never acquired more than that.

Reply:
Yes that is fine!